components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -706,47 +732,231 @@ public static String getUserConsentURL(OAuth2Parameters params, String loggedInU
+                    "Error occurred while removing OIDC scopes from disapproved OAuth scopes.", e);
+        } catch (IdentityOAuth2ScopeException e) {
+            throw new OAuthSystemException("Error occurred while storing OAuth scope consent.", e);
+        }
+    }
+
+    private static List<String> getAllowedOAuthScopes(OAuth2Parameters params) throws OAuthSystemException {
+
+        Set<String> allowedScopes = params.getScopes();
+        List<String> allowedOAuthScopes = new ArrayList<>();
+        if (CollectionUtils.isNotEmpty(allowedScopes)) {
+            try {
+                startTenantFlow(params.getTenantDomain());
 
             /* If DropUnregisteredScopes scopes config is enabled
              then any unregistered scopes(excluding internal scopes
              and allowed scopes) is be dropped. Therefore they will
              not be shown in the user consent screen.*/
-            if (oauthServerConfiguration.isDropUnregisteredScopes()) {
-                if (log.isDebugEnabled()) {
-                    log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+                if (oauthServerConfiguration.isDropUnregisteredScopes()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+                    }
+                    allowedScopes = dropUnregisteredScopes(params);
                 }
-                allowedScopes = dropUnregisteredScopes(params);
+
+                // Get registered OIDC scopes.
+                String[] oidcScopes = oAuthAdminService.getScopeNames();
+                List<String> oidcScopeList = new ArrayList<>(Arrays.asList(oidcScopes));
+                for (String scope : allowedScopes) {
+                    if (!oidcScopeList.contains(scope)) {
+                        allowedOAuthScopes.add(scope);
+                    }
+                }
+            } catch (IdentityOAuthAdminException e) {
+                throw new OAuthSystemException("Error while retrieving OIDC scopes.", e);
+            } finally {
+                PrivilegedCarbonContext.endTenantFlow();
             }
+        }
+        return allowedOAuthScopes;
+    }
 
-            // Get registered OIDC scopes.
-            String[] oidcScopes = oAuthAdminService.getScopeNames();
-            List<String> oidcScopeList = new ArrayList<>(Arrays.asList(oidcScopes));
-
-            /* Remove OIDC scopes from scope string sent back
-             in the consent page as consent is not needed for
-             OIDC scopes.*/
-            StringBuilder scopes = new StringBuilder();
-            for (String scope : allowedScopes) {
-                if (!oidcScopeList.contains(scope)) {
-                    scopes.append(scope).append(" ");
+    private static String getConsentRequiredScopes(AuthenticatedUser user, OAuth2Parameters params)
```

Comment:
This method seems to be doing two things 

1. Setting consentRequiredScopes required scopes to oauth params
2. Sending back consentRequiredScopes

If the method is refactored then the places that depend on setting consentRequiredScopes oauth param will break.
since there is only one actual usage this method as String: getConsentRequiredScopes(..)  lets rename the method to reflect setConsentRequiredScopesToOauthParams and then in the usage retrieve the scopes from the oauth params

Reply:
Fixed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1548/commits/27ddbc8fed9c4368ffe7a826b857537a6b8ffff0